build-infra/build-vm.sls

```diff
@@ -206,29 +206,22 @@ github.com:
 {% endif %}
 
 {% for builder in builders_list %}
-{% set config_baseurl = salt['pillar.get']('build-infra:build-envs:' + env + ':builders-list:' + builder + ':config:repository:baseurl', 'https://github.com/QubesOS/qubes-') %}
+{% set config_baseurl = salt['pillar.get']('build-infra:build-envs:' + env + ':builders-list:' + builder + ':config:repository:baseurl', 'ssh://github.com/QubesOS/qubes-') %}
 {% set config_repo = salt['pillar.get']('build-infra:build-envs:' + env + ':builders-list:' + builder + ':config:repository:component', 'release-configs') %}
 {% set config_file = salt['pillar.get']('build-infra:build-envs:' + env + ':builders-list:' + builder + ':config:file') %}
 {% set keys =  salt['pillar.get']('build-infra:build-envs:' + env + ':builders-list:' + builder + ':keys', []) %}
 
-{{builder}}-get:
-  git.latest:
-    - name : https://github.com/QubesOS/qubes-builder
-    - target: {{ builder }}
-    - user: user
```

Comment:
I see [`git.latest` has `rev`](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.git.html#salt.states.git.latest) which seems to do almost exactly what we need.
I don't see an option to skip this if it's already cloned (specifically: avoid rollback, and do not report a failure).

Reply:
I am not sure if `rev` does what we want.  For example, I am not sure if it passes `--no-checkout` to `git clone`.